File: Data-Structure/Stacks/Simple Text Editor.py
c = '' 
stk=[]
stk.append(c)
for _ in range(int(input())):
    l = input().split()
    if l[0] == "1":
        c+=l[1]
        stk.append(c)
    elif l[0] == "2":
        c = c[0:len(c)-int(l[1]):]
        stk.append(c)
    elif l[0] == "3":
        print(c[int(l[1])-1])
    else:
        stk.pop()
        c = stk[len(stk)-1]

File: Data-Structure/Stacks/Simple Text Editor.cpp
#include <bits/stdc++.h>
using namespace std;
#define int int64_t

signed main(void){
    ios::sync_with_stdio(false);
    cin.tie(nullptr); 
    cout.tie(nullptr);
#ifdef HELL_JUDGE
    freopen("input","r",stdin);
    freopen("output","w",stdout);
#endif
    int q,op,k; cin >> q; 
    string c{},temp{};
    stack<string>s;
    s.push(c);
    while (q--) {
        cin >> op; 
        switch(op) {
            case 1:
                cin >> temp; 
                c+=temp; 
                s.push(c);
                break; 
            case 2:
                cin >> k; 
                c = c.substr(0,c.length()-k); 
                s.push(c);
                break;
            case 3:
                cin >> k; 
                cout << c[k-1] << '\n'; 
                break;
            case 4:
                s.pop(); 
                c = s.top();
                break;
        }
    }
    return 0;
}
